plugins/prmctl/src/main/java/kg/apc/jmeter/control/sampler/SetVariablesAction.java

```diff
@@ -0,0 +1,52 @@
+package kg.apc.jmeter.control.sampler;
+
+import org.apache.jmeter.config.Arguments;
+import org.apache.jmeter.samplers.AbstractSampler;
+import org.apache.jmeter.samplers.Entry;
+import org.apache.jmeter.samplers.SampleResult;
+import org.apache.jmeter.testelement.property.JMeterProperty;
+import org.apache.jmeter.testelement.property.TestElementProperty;
+import org.apache.jmeter.threads.JMeterContextService;
+import org.apache.jmeter.threads.JMeterVariables;
+import org.apache.jorphan.logging.LoggingManager;
+import org.apache.log.Logger;
+
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ *
+ */
+public class SetVariablesAction extends AbstractSampler {
+    private static final Logger log = LoggingManager.getLoggerForClass();
```

Comment:
Because we want to be compatible with older versions of JMeter

Reply:
Hello Andrey,
How far ? for how much time ? 
slf4j is available in dependencies of JMeter since 3.0.

Those old versions are bad for JMeter reputation and they suffer now from 2 identified CVEs in remote testing.

Reply:
I don't follow the opinion of JMeter project here, sorry. I follow what I hear from users, which versions they still actively use. I know that 2.13 is the lowest required version for now. It tends to slowly go away, but takes time... Maybe a year from now we'll be safe to refuse 2.13, but not today.

Reply:
From your project stats, 2.13 represent 2% of users of jmeter-plugins.

Shouldn't your policy be,  "You want new features, upgrade plugins+jmeter" ?

But of course, you're free :-)

Reply:
Again, the policy is not to dictate users what they have to do. They don't like blackmail.
Instead, I want to let users more freedom of choice.

Regarding stats, it's not only quantity of users, it's also quality. I know that some important companies still use 2.13 for their own reasons. I respect those rare, but important users.

I am also feeling safe by the fact JMeter still supports compatibility with old versions around logging. In my opinion it's cheap and user-friendly.

Reply:
I love users as much as you do.
It's not blackmail :-) , it's just that OSS takes time, handling backward compatibility takes time, keeping old stack can be deadly.
Ok for me anyway.

PS : you could add slf4j as dependency of plugin, it would work even with 2.X.